helsingborg-stad/helsingborg-io-sls-api
# feat: add lambda for checking the status of a users application version

## Explain the changes you’ve made
Added a new api (version-api) for fetching a status if a users application version of Mitt Helsingborg needs to be updated, can be updated or if update is required.

## Explain why these changes are made
To make sure that a user of Mitt Helsingborg is using an application version that our backend supports, we want to check that the version used is within `min` and `max` version numbers configured in our backend.

If a version is below `min` we want to force the user to update the application.
If a version is between `min` and `max` we (possibly) want to recommend the user to update.
If a version is the same as `max`, then everything is OK.

## Explain your solution
The lambda function fetches the configurations from the SSM parameter store and (which is manually configured) and checks the `User-Agent` header sent from the application. From the `User-Agent` header, the application retrieves the device OS and application version used by that device and calculates the status depending on the SSM paramters.

## How to test

1. Checkout this branch
2. run `sls deploy` within service folder for service `status-api`
3. Make sure you have `version` gateway path deployed in resources repo
4. Make sure that you have SSM Parameter store values deployed which follows the format in this clickup: https://app.clickup.com/t/1few6ak
5. Run the endpoint from postman or application

## Tested environments

- [] Your local Serverless environment.
